app/models/user.rb

```diff
@@ -3,13 +3,34 @@ class User < ApplicationRecord
 
   validates :name, presence: true, uniqueness: true
   validates :email, uniqueness: true, format: {with: URI::MailTo::EMAIL_REGEXP}
+  validates :email, uniqueness: true
   validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }
 
+  after_destroy :ensure_an_admin_remains
+  after_create_commit :welcome_mail
+  before_update :ensure_not_admin
+  before_destroy :ensure_not_admin
```

Comment:
Make the methods called here as `private`

Reply:
should i make them private individually?

Reply:
Personal preference to use `private` as block or make methods `private` indiviually
You can go with any one

Reply:
ok